lib/twilio/device.ts

```diff
@@ -1430,6 +1449,7 @@ class Device extends EventEmitter {
       audioContext: Device.audioContext,
       audioProcessorEventObserver: this._audioProcessorEventObserver,
       enumerateDevices: this._options.enumerateDevices,
+      getMakeCallPromise: this.getMakeCallPromise.bind(this),
```

Comment:
If you consider my other comments, this would become
```ts
beforeSetInputDevice: () => {
  if (this._makeCallPromise) {
    this._log.debug('beforeSetInputDevice pause detected');
    return this._makeCallPromise;
  } else {
    this._log.debug('beforeSetInputDevice pause not detected, setting default');
    return Promise.resolve();
  }
}
```